examples/api/lib/material/page_transitions_theme/page_transitions_theme.0.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for [PageTransitionsTheme].
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const PageTransitionsThemeApp());
+
+class PageTransitionsThemeApp extends StatelessWidget {
+  const PageTransitionsThemeApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(
+        useMaterial3: true,
+        // Defines the page transition animations used by MaterialPageRoute
+        // for different target platforms.
+        // Non-specified target platforms will default to
+        // ZoomPageTransitionsBuilder().
+        pageTransitionsTheme: const PageTransitionsTheme(
+          builders: <TargetPlatform, PageTransitionsBuilder>{
+            TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
+            TargetPlatform.linux: OpenUpwardsPageTransitionsBuilder(),
+            TargetPlatform.macOS: FadeUpwardsPageTransitionsBuilder(),
+            // The snapshotting optimization for entering route can be
```

Comment:
It's a little confusing to say that we're deactivating the snapshotting optimzation, since presumably we're turning allowEnterRouteSnapshotting off as a performance optimization. It might be simpler to demo the allowSnapshotting property. You could reuse the API doc's description for the comment: Improve performance by disabling [SnapshotWidget]-based animations on both the outgoing and incoming route.

Reply:
> since presumably we're turning allowEnterRouteSnapshotting off as a performance optimization

It’s very interesting that you made this comment because it shows that the documentation is not clear enough.
Enabling snapshotting is the performance optimization. The main documentation for `allowSnapshotting` explains this well (https://master-api.flutter.dev/flutter/widgets/PageRoute/allowSnapshotting.html) but the `_ZoomPageTransition.allowSnapshotting` comment is ambiguous.

I updated the PR with a change in _ZoomPageTransition.allowSnapshotting comment.

> It might be simpler to demo the allowSnapshotting property

I hesitated to demo this in the example, but it turned out that we had several reports related to this (for instance  https://github.com/flutter/flutter/issues/121100 and https://github.com/flutter/flutter/issues/119897). And one goal was to have somewhere to point out.

I too would have preferred to demo `allowSnapshotting` instead of `allowEnterRouteSnapshotting` but allowSnapshotting is not available on ZoomPageTransitionsBuilder.

It would be great to discuss if we want to expose this property. My concern is that exposing it might lead to people deactivating the optimization just in case…
While looking again at ZoomPageTransitionsBuilder, I think not exposing this property was done on purpose and that it is expected to configure this only on a per route basis instead of configuring this globally.

Demoing neither `allowSnapshotting` nor `allowEnterRouteSnapshotting` in this PR code sample seems to be the best solution for the moment. It is probably wiser to demo ‘allowSnapshotting’ by adding an example to `MaterialPageRoute`.

Reply:
Its true that globally disabling this property is not easy right now. I would not be opposed to making it public though - if folks want to opt out they should certainly be able to